<?xml version="1.0"?>
<launch>
	<arg name="prefix" default="summit_xl_"/>
    
    <!-- Pixhawk -->
	<arg name="prefix" default="summit_xl_"/>
    <arg name="fcu_url" default="/dev/ttyUSB_PX4:921600" />
    <arg name="gcs_url" default="" />
    <arg name="tgt_system" default="1" />
    <arg name="tgt_component" default="1" />
    <arg name="log_output" default="screen" />
    <arg name="imu_frame_id" default="imu_link" />
	<arg name="pluginlists_yaml" default="$(find whs_summit)/config/mavros/px4_pluginlists.yaml" />
	<arg name="config_yaml" default="$(find mavros)/launch/px4_config.yaml" />

	<node pkg="mavros" type="mavros_node" name="mavros" respawn="false" clear_params="true" output="$(arg log_output)">
		<param name="fcu_url" value="$(arg fcu_url)" />
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="target_system_id" value="$(arg tgt_system)" />
		<param name="target_component_id" value="$(arg tgt_component)" />

		<!-- load blacklist, config -->
		<rosparam command="load" file="$(arg pluginlists_yaml)" />
		<rosparam command="load" file="$(arg config_yaml)" />
		<param name="~imu/frame_id" value="$(arg prefix)$(arg imu_frame_id)" />
	</node>

    <node pkg="imu_complementary_filter" type="complementary_filter_node" name="complementary_filter_node" output="screen">
        <remap from="imu/data_raw" to="mavros/imu/data_raw"/>
        <remap from="imu/mag" to="mavros/imu/mag"/>
        <remap from="imu/data" to="summit_xl/imu/data"/>
        <param name="fixed_frame" value="$(arg prefix)imu_link"/>
        <param name="publish_tf" value="false"/>
        <param name="publish_debug_topics" value="true"/>
        <param name="do_bias_estimation" value="false"/>
        <param name="do_adaptive_gain" value="true"/>
        <param name="use_mag" value="true"/>
        <param name="gain_acc" value="0.005"/>
        <param name="gain_mag" value="0.001"/>
    </node>

    <node pkg="robot_localization" type="ekf_localization_node"
        name="robot_localization_ekf_node_odom"
        clear_params="true">
        <param name="frequency" value="10" />  
        <param name="sensor_timeout" value="0.2" />  
        <param name="two_d_mode" value="true" />
        <param name="publish_tf" value="false" />
        <param name="map_frame" value="summit_xl_map" />
        <param name="odom_frame" value="summit_xl_odom2" />
        <param name="base_link_frame" value="summit_xl_base" />
        <param name="world_frame" value="summit_xl_odom2" />
        <param name="imu0" value="mavros/imu/data"/>
        <rosparam param="imu0_config">[false, false, false,
                                    false, false, true,
                                    false, false, false,
                                    false, false, true,
                                    true, false, false]</rosparam>
        <param name="odom0" value="summit_xl/robotnik_base_control/odom"/>
        <rosparam param="odom0_config">[true,  true,  false,
                                false, false, true,
                                true,  false, false,
                                false, false, true,
                                false, false, false]</rosparam>
    </node>

</launch>